server/etcdserver/api/v3rpc/metrics.go

```diff
@@ -58,3 +65,46 @@ func init() {
 	prometheus.MustRegister(streamFailures)
 	prometheus.MustRegister(clientRequests)
 }
+
+func splitMethodName(fullMethodName string) (string, string) {
+	fullMethodName = strings.TrimPrefix(fullMethodName, "/") // remove leading slash
+	if i := strings.Index(fullMethodName, "/"); i >= 0 {
+		return fullMethodName[:i], fullMethodName[i+1:]
+	}
+	return "unknown", "unknown"
+}
+
+// constructExtensiveMetricsInterceptors constructs unary and stream interceptors to record histogram metrics for gRPC requests
+func constructExtensiveMetricsInterceptors() (grpc.UnaryServerInterceptor, grpc.StreamServerInterceptor) {
```

Comment:
We don't need this function anymore with the patch in https://github.com/etcd-io/etcd/pull/19195#issuecomment-2624840140

Reply:
removed